File: leetCode/fukun/201-300/lc_230.js
var kthSmallest = function (root, k) {
    let n = 0;
    const helper = function (node) {
        if (!node) return;
        let l = helper(node.left)
        if (l || l === 0) return l;
        n++;
        if (n === k) return node.val;
        let r = helper(node.right);
        if (r || r === 0) return r;
    }
    return helper(root);
};

File: leetCode/lwc/0201-0300/lc_230.java
package leetcode;

import java.util.Comparator;
import java.util.PriorityQueue;

/**
 * @author lwc
 * @date 2019/11/25 9:28
 */
public class lc_230 {
    public int kthSmallest(TreeNode root, int k) {
        PriorityQueue<Integer> queue = new PriorityQueue<>(k, Comparator.reverseOrder());
        recursive(root, k, queue);
        return queue.peek();
    }

    public void recursive(TreeNode root, int k, PriorityQueue queue){
        if(root == null) return;
        recursive(root.left, k, queue);
        if(queue.size() < k)
            queue.offer(root.val);
        recursive(root.right, k, queue);
    }
}
